pontoon/sync/tasks.py

```diff
@@ -247,18 +218,30 @@ def sync_translations(
             pk__in=changed_locales_pks + readonly_locales_pks
         )
 
-    # Pull VCS changes in case we're on a different worker than the one
-    # sync started on.
     if not no_pull:
-        log.info("Pulling changes for project {0} started.".format(db_project.slug))
-        repos_changed, repo_locales = pull_changes(db_project, locales)
-        repos = repos.filter(pk__in=repo_locales.keys())
-        log.info("Pulling changes for project {0} complete.".format(db_project.slug))
+        repo_locales = {db_project.source_repository.pk: Locale.objects.none()}
+
+        # Pull repos of locales in case of multi_locale_project
+        if not db_project.has_single_repo:
+            log.info(
+                "Pulling locale repos for project {0} started.".format(db_project.slug)
+            )
+            locale_repos_changed, not_pulled_repo_locales = pull_locale_repo_changes(
```

Comment:
Why `not_pulled_repo_locales`, if they are actually pulled?

`not_yet_pulled_repo_locales` would be more accurate, but that's pretty long. Simply `pulled_repo_locales` will do.